apps/classroomio-com/src/lib/Navigation/Navigation.svelte

```diff
@@ -32,13 +32,29 @@
 
   function scroll(key = '') {
     const el = document.getElementById(key);
+    console.log('Scrolling to element:', el);
     setTimeout(() => {
-      el?.scrollIntoView({
-        behavior: 'smooth',
-        block: 'start',
-        inline: 'nearest'
-      });
-    }, 50);
+      if (el) {
+        el.scrollIntoView({
+          behavior: 'smooth',
+          block: 'start',
+          inline: 'nearest'
+        });
+      } else {
+        // goto('/') if element not found, then try scrolling again
+        goto('/');
+        setTimeout(() => {
+          const elAfterGoto = document.getElementById(key);
+          if (elAfterGoto) {
+            elAfterGoto.scrollIntoView({
+              behavior: 'smooth',
+              block: 'start',
+              inline: 'nearest'
+            });
+          }
+        }, 200);
+      }
+    }, 200);
```

Comment:
@tunny17 why is there a setTimeout in a setTimeout in another setTimeout?

How does this work actually.

Note there one setTimeout in line 117, then another on line36, then another in line 46.

I think it should be something like:

```js
const scroll = () => {
  const el = document.getElementById(key);
  if (el) {
    setTimeout(() => {
      el.scrollIntoView({
          behavior: 'smooth',
          block: 'start',
          inline: 'nearest'
        });
    }, 200)
  } else {
    goto('/');
    // ... the other code here
  }
}

// even listener here
on:m-click={scroll}
```

Reply:
lol, fixed